core/coreapi/interface/options/unixfs.go

```diff
@@ -42,7 +42,15 @@ type UnixfsAddSettings struct {
 	Progress bool
 }
 
+type UnixfsLsSettings struct {
+	Async bool
```

Comment:
So, what if we just *always* streamed at the CoreAPI layer and collected/sorted the results at the commands layer?

Reply:
Good point, 40 lines less to care about